### PR TITLE
Add AOT + HybridSim tests for MaxText v4 configs

### PR DESCRIPTION
**Pull request by raymondzouu**
# Description

Add AOT + HybridSim tests for MaxText v4 configs

# Tests

Tested in local Composer env: http://shortn/_ILuRzE2MUb

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [x] I have performed a self-review of my code.
- [x] I have necessary comments in my code, particularly in hard-to-understand areas.
- [x] I have run one-shot tests and provided workload links above if applicable. 
- [x] I have made or will make corresponding changes to the doc if needed.

### CODE REVIEW

File: dags/multipod/maxtext_v4_configs_aot_hybridsim.py [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+          maxtext_hybridsim = gke_config.get_gke_config(
+              tpu_version=tpu,
+              tpu_cores=cores,
+              tpu_zone=zone,
+              project_name=project_name,
+              cluster_name=cluster_name,
+              time_out_in_min=240,
+              test_name=f"maxtext-{model_size}-{n}xv{tpu.value}-{num_cores}-hybridsim",
+              run_model_cmds=hybridsim_cmd,
+              docker_image="gcr.io/tpu-prod-env-multipod/internal_cloud_hybridsim_nightly:2024-04-18",
```
Comment (jonb377) [inline dags/multipod/maxtext_v4_configs_aot_hybridsim.py:105]: The date is pinned to 04/18, do you want to make this a field in `DockerImage` like the MaxText images?

File: dags/multipod/maxtext_v4_configs_aot_hybridsim.py [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+              docker_image=DockerImage.MAXTEXT_TPU_JAX_STABLE.value,
+              test_owner=test_owner.RAYMOND_Z,
+          ).run(gcs_location=shared_gcs_location)
+
+          # Run HybridSim workload: read HLO from GCS, generate estimated step time
+          cluster_name, zone, project_name, cores = clusters[tpu]
+          chip_config = "default" if tpu == TpuVersion.V5E else "megacore"
+          hybridsim_cmd = (
+              "gsutil cp gs://cloud-hybridsim-prod/run_hybridsim.sh .",
+              f"bash run_hybridsim.sh GCS_PATH=${{GCS_OUTPUT}}xla_dump CHIP_CONFIG={chip_config}",
```
Comment (jonb377) [inline dags/multipod/maxtext_v4_configs_aot_hybridsim.py:94]: How are we processing the resulting step time estimation?

File: dags/multipod/maxtext_v4_configs_aot_hybridsim.py [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+from airflow.utils.task_group import TaskGroup
+from dags.multipod.configs import gke_config
+
+# Run once a day at 10 am UTC (2 am PST / 3 am PDT)
+SCHEDULED_TIME = "0 10 * * *" if composer_env.is_prod_env() else None
+
+with models.DAG(
+    dag_id="maxtext_v4_configs_aot_hybridsim",
+    schedule=SCHEDULED_TIME,
+    tags=["multipod_team", "maxtext", "stable", "nightly"],
```
Comment (jonb377) [inline dags/multipod/maxtext_v4_configs_aot_hybridsim.py:33]: We cna drop the `nightly` tag, since this only uses the stable image

**Review by jonb377 — APPROVED** [review]
LGTM, nice work Raymond!

**Review by tonyjohnchen — APPROVED** [review]
LGTM!
Thanks!

File: dags/multipod/maxtext_v4_configs_aot_hybridsim.py (hunk 1 above)
Comment (raymondzouu) [reply dags/multipod/maxtext_v4_configs_aot_hybridsim.py:105]: This is a temporary docker image that we've manually built. I'm thinking that when we have the automatic nightly hybridsim image set up I will add it to `DockerImage`.

File: dags/multipod/maxtext_v4_configs_aot_hybridsim.py (hunk 2 above)
Comment (raymondzouu) [reply dags/multipod/maxtext_v4_configs_aot_hybridsim.py:94]: This is still in progress. The step time estimation is printed to logs and we will need to capture it and write it to a jsonlines file for XLML to parse.

File: dags/multipod/maxtext_v4_configs_aot_hybridsim.py (hunk 3 above)
Comment (raymondzouu) [reply dags/multipod/maxtext_v4_configs_aot_hybridsim.py]: Done.

File: dags/multipod/maxtext_v4_configs_aot_hybridsim.py (hunk 1 above)
Comment (tonyjohnchen) [reply dags/multipod/maxtext_v4_configs_aot_hybridsim.py:105]: We are working on the nightly release. 
We will update this once it is ready.

**Pull request merged**